### ISSUE TITLE
An error occurred while Rails/FilePath cop was inspecting String interpolation containing Rails.root

### ISSUE THREAD

**Issue by liaham** [rubocop/rubocop#12119]
*Be clear, concise and precise in your description of the problem.
Open an issue with a descriptive title and a summary in grammatically correct,
complete sentences.*

*Use the template below when reporting bugs. Please, make sure that
you're running the latest stable RuboCop and that the problem you're reporting
hasn't been reported (and potentially fixed) already.*

*Before filing the ticket you should replace all text above the horizontal
rule with your own words.*

--------

## Expected behavior

I expect rubocop to inspect String interpolation containing `Rails.root` without any error. This is the line causing the error:

```ruby
msg = "Navigate to #{Rails.root}. Position is: #{Dir.pwd}"
```

Please note, the line belongs to a plugin located outside of a rails app when rubocop runs over it. 

I expected this behavior because it was no problem in the past. Please don't ask me for the last rubocop version causing no error like that above. I don't know.  

## Actual behavior

```ruby
An error occurred while Rails/FilePath cop was inspecting /home/path/to/file/data.rb:128:16.
undefined method `str_type?' for nil:NilClass

          return unless rails_root_node.arguments.last.str_type?
                                                      ^^^^^^^^^^
/home/of/user/.rbenv/versions/3.1.4/lib/ruby/gems/3.1.0/gems/rubocop-rails-2.20.2/lib/rubocop/cop/rails/file_path.rb:166:in `autocorrect_extension_after_rails_root_join_in_dstr'
```

## Steps to reproduce the problem

1. Create a directory with a ruby file containing a line like that above. 
2. Create also `.rubocop.yml` with the following content:

```yaml
---
inherit_from: .rubocop_todo.yml

AllCops:
  NewCops: enable
  DisplayCopNames: true
  DisplayStyleGuide: true
  TargetRubyVersion: 2.7

  Exclude:
    - '**/vendor/**/*'
    - '**/tmp/**/*'
    - '**/bin/**/*'
    - '**/extra/**/*'
    - '**/lib/generators/**/templates/*'
    - '**/lib/tasks/**/*'
    - '**/files/**/*'
    - '**/test/**/*'
    - '**/db/migrate/*'
    - 'db/schema.rb'
    - '**/*_patch.rb'

require:
  - rubocop-performance
  - rubocop-rails

Style/FrozenStringLiteralComment:
  Enabled: true
  EnforcedStyle: always
  Exclude:
    - 'db/**/*.rb'
    - 'Gemfile'
    - 'Rakefile'

Metrics/AbcSize:
  Exclude:
    - 'test/**/*'

Metrics/MethodLength:
  Exclude:
    - 'test/**/*'

Layout/LineLength:
  Max: 120

Style/Documentation:
  Enabled: false

Style/Encoding:
  Enabled: false

Layout/TrailingWhitespace:
  AllowInHeredoc: true

Rails/ApplicationRecord:
  Enabled: false

Naming/VariableNumber:
  Exclude:
    - 'test/**/*'
  
Rails/ActionControllerFlashBeforeRender:
 Enabled:   false

Rails/OutputSafety:
  Exclude:
    - 'app/presenters/**/*'
    - 'app/helpers/**/*'
```

3. Install:
   -  ruby 3.1.4 
   -  rubocop 1.56.0 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.1.4) [x86_64-linux]
   - rubocop-rails 2.20.2

## RuboCop version

```
$ rubocop -V
1.56.0 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.1.4) [x86_64-linux]
  - rubocop-performance 1.18.0
  - rubocop-rails 2.20.2
```
